src/ppx/xr_component.cpp

```diff
@@ -417,6 +504,45 @@ void XrComponent::PollEvents(bool& exitRenderLoop)
                 break;
         }
     }
+
+    XrActiveActionSet activeActionSet{mImguiInput, XR_NULL_PATH};
+    XrActionsSyncInfo syncInfo{XR_TYPE_ACTIONS_SYNC_INFO};
+    syncInfo.countActiveActionSets = 1;
+    syncInfo.activeActionSets      = &activeActionSet;
+    XrResult syncResult            = xrSyncActions(mSession, &syncInfo);
+    if (syncResult == XR_SESSION_NOT_FOCUSED) {
+        return;
+    }
+    CHECK_XR_CALL(syncResult);
+
+    XrActionStateBoolean clickActionState    = {XR_TYPE_ACTION_STATE_BOOLEAN};
+    XrActionStatePose    pointingActionState = {XR_TYPE_ACTION_STATE_POSE};
+
+    XrActionStateGetInfo getInfo{XR_TYPE_ACTION_STATE_GET_INFO};
+    getInfo.action = mImguiClickAction;
+    CHECK_XR_CALL(xrGetActionStateBoolean(mSession, &getInfo, &clickActionState));
+    getInfo.action = mImguiPointingAction;
+    CHECK_XR_CALL(xrGetActionStatePose(mSession, &getInfo, &pointingActionState));
+
+    if (clickActionState.isActive) {
+        mImguiClickState = clickActionState.currentState;
+    }
+    if (pointingActionState.isActive) {
+        XrSpaceLocation pointingLocation{XR_TYPE_SPACE_LOCATION};
+        XrResult        locateResult = xrLocateSpace(mImguiPointingSpace, mUISpace, mImguiActionTime, &pointingLocation);
+        if (locateResult != XR_SESSION_NOT_FOCUSED) {
+            CHECK_XR_CALL(locateResult);
+            mImguiPointingState = pointingLocation.pose;
+        }
+    }
+}
+
+std::optional<XrVector2f> XrComponent::GetUICursor() const
+{
+    if (!mImguiPointingState.has_value()) {
+        return std::nullopt;
+    }
+    return ProjectCursor(mImguiPointingState.value(), -0.5f);
```

Comment:
out of curiosity, why is the z -0.5f?

Reply:
That's where the layer is, see `ConditionallyPopulateImGuiLayer`. Added a constant for it.